endpoints/endpoints.py

```diff
@@ -0,0 +1,1427 @@
+import calendar
+from fabric import Connection
+from invoke import run
+import json
+import logging
+import os
+from pathlib import Path
+import re
+import sys
+import time
+
+TOOLBOX_HOME = os.environ.get('TOOLBOX_HOME')
+if TOOLBOX_HOME is None:
+    print("This script requires libraries that are provided by the toolbox project.")
+    print("Toolbox can be acquired from https://github.com/perftool-incubator/toolbox and")
+    print("then use 'export TOOLBOX_HOME=/path/to/toolbox' so that it can be located.")
+    exit(1)
+else:
+    p = Path(TOOLBOX_HOME) / 'python'
+    if not p.exists() or not p.is_dir():
+        print("ERROR: <TOOLBOX_HOME>/python ('%s') does not exist!" % (p))
+        exit(2)
+    sys.path.append(str(p))
+from toolbox.json import *
+
+ROADBLOCK_HOME = os.environ.get('ROADBLOCK_HOME')
+if ROADBLOCK_HOME is None:
+    print("This script requires libraries that are provided by the roadblock project.")
+    print("Roadblock can be acquired from https://github.com/perftool-incubator/roadblock and")
+    print("then use 'export ROADBLOCK_HOME=/path/to/roadblock so that it can be located.")
+    exit(1)
+else:
+    p = Path(ROADBLOCK_HOME) / 'roadblock.py'
+    if not p.exists() or not p.is_file():
+        print("ERROR: <ROADBLOCK_HOME>/roadblock.py ('%s') does not exist!" % (p))
+        exit(2)
+    sys.path.append(str(Path(ROADBLOCK_HOME)))
+from roadblock import roadblock
+
+roadblock_exits = {
+    "success": 0,
+    "timeout": 3,
+    "abort": 4,
+    "input": 2
+}
+
+log = logging.getLogger(__file__)
+
+def run_remote(connection, command, validate = False, debug = False):
+    """
+    Run a command on a remote server using an existing Fabric connection
+
+    Args:
+        connection (Fabric Connection): The connection to use to run the command remotely
+        command (str): The command to run
+        validate (bool): Is the function being called from validation mode (which means that logging cannot be used)
+        debug (bool): Is debug output enabled during validation mode
+
+    Globals:
+        None
+
+    Returns:
+        a Fabric run result
```

Comment:
I like the docstring for the functions. Fabric result is int ? We could adopt (type) for the return as well.

Reply:
It is not an int, it is an object with multiple properties:

https://docs.pyinvoke.org/en/latest/api/runners.html#invoke.runners.Result